tests/test_nn/test_emb.py

```diff
@@ -113,6 +113,30 @@ def _pre_instantiation_hook(self, kwargs: MutableMapping[str, Any]) -> MutableMa
         return kwargs
 
 
+class NodePieceTests(cases.RepresentationTestCase):
+    """Tests for node piece representation."""
+
+    cls = pykeen.nn.emb.NodePieceRepresentation
+    num_entities: int = 8
+    num_relations: int = 7
+    num_triples: int = 31
+    kwargs = dict(
+        token_representation=pykeen.nn.emb.EmbeddingSpecification(
+            shape=(3,),
+        )
+    )
+
+    def _pre_instantiation_hook(self, kwargs: MutableMapping[str, Any]) -> MutableMapping[str, Any]:
+        kwargs = super()._pre_instantiation_hook(kwargs)
+        kwargs["triples_factory"] = generate_triples_factory(
+            num_entities=self.num_entities,
+            num_relations=self.num_relations,
+            num_triples=self.num_triples,
+            create_inverse_triples=False,
+        )
```

Comment:
This is duplicated code to RGCNRepresentation tests

Reply:
It's sort of duplicated, but abstracting this would actually create more code at this point. I'll leave a note for later, just in case.

Reply:
TODO added (for a later PR) in 7e1813ef